plugins/catalog-react/src/components/EntityNamespacePicker/EntityNamespacePicker.tsx

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity } from '@backstage/catalog-model';
+import {
+  Box,
+  Checkbox,
+  FormControlLabel,
+  makeStyles,
+  TextField,
+  Typography,
+} from '@material-ui/core';
+import CheckBoxIcon from '@material-ui/icons/CheckBox';
+import CheckBoxOutlineBlankIcon from '@material-ui/icons/CheckBoxOutlineBlank';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import { Autocomplete } from '@material-ui/lab';
+import React, { useEffect, useMemo, useState } from 'react';
+import { useEntityList } from '../../hooks/useEntityListProvider';
+import { EntityNamespaceFilter } from '../../filters';
+
+/** @public */
+export type CatalogReactEntityNamespacePickerClassKey = 'input';
+
+const useStyles = makeStyles(
+  {
+    input: {},
+  },
+  {
+    name: 'CatalogReactEntityNamespacePicker',
+  },
+);
+
+const icon = <CheckBoxOutlineBlankIcon fontSize="small" />;
+const checkedIcon = <CheckBoxIcon fontSize="small" />;
+
+/** @public */
+export const EntityNamespacePicker = (props: { initialFilter?: string[] }) => {
+  const { initialFilter = [] } = props;
+  const classes = useStyles();
+  const {
+    updateFilters,
+    backendEntities,
+    filters,
+    queryParameters: { namespace: namespaceParameter },
+  } = useEntityList();
+
+  const queryParamNamespace = useMemo(
+    () => [namespaceParameter].flat().filter(Boolean) as string[],
+    [namespaceParameter],
+  );
+
+  const [selectedNamespace, setSelectedNamespace] = useState(
```

Comment:
```suggestion
  const [selectedNamespaces, setSelectedNamespaces] = useState(
```
I'd switch this, `queryParamNamespace` and `availableNamespace` to plural for better code intuition at a glance, since these are arrays.